webui/react/src/pages/Settings/UserSettings.settings.ts

```diff
@@ -0,0 +1,28 @@
+import { any } from 'io-ts';
+
+import { SettingsConfig } from 'hooks/useSettings';
+import { KeyboardShortcut } from 'utils/shortcut';
+
+export interface Settings {
+  navbarCollapsed: KeyboardShortcut;
+}
+
+const shortCutSettingsConfig: SettingsConfig<Settings> = {
+  settings: {
+    navbarCollapsed: {
+      defaultValue: {
+        alt: false,
+        ctrl: false,
+        key: 'u',
+        meta: true,
+        shift: true,
+      },
+      skipUrlEncoding: true,
+      storageKey: 'navbarCollapsed',
+      type: any,
```

Comment:
Yea..  I should have asked about this explicitly.. I was thinking about define something like this but I was not sure..
```
const keyboardShortcutType = new t.Type<KeyboardShortcut, KeyboardShortcut, unknown>(
  'KeyboardShortcut',
  (input: unknown): input is KeyboardShortcut => typeof input === 'object',
  // `t.success` and `t.failure` are helpers used to build `Either` instances
  (input, context) => (instanceOfShortcut(input) ? t.success(input) : t.failure(input, context)),
  // `A` and `O` are the same, so `encode` is just the identity function
  t.identity,
);
```

Reply:
You don't need to define a type from scratch, you can use `t.type` ([example here](https://github.com/determined-ai/determined/blob/main/webui/react/src/ioTypes.ts#L69))

Reply:
Also recommend using the same name:
```typescript
export const Shortcut = t.type({
  meta: t.boolean,
  ...
  key: t.string
})
export type Shortcut = t.TypeOf<typeof Shortcut>
```

Reply:
I see! Thanks for the help! Appreciate it! 
Since I picked up another ticket related to keyboard shortcuts, should I apply these changes within that ticket, or make a separate PR to fix it?

Reply:
You can roll it into that PR